nvflare/private/fed/utils/identity_utils.py

```diff
@@ -58,6 +59,27 @@ def load_cert_bytes(data: bytes):
     return load_crt_bytes(data)
 
 
+def get_parent_site_name(fqsn: str) -> Optional[str]:
+    """Get the parent site's name for a specified FQSN (fully qualified site name)
+
+    Args:
+        fqsn: the FQSN to find parent for
+
+    Returns: the parent site's name or None if the FQSN doesn't have a parent
+
+    """
+    if not fqsn:
+        return None
+
+    if not isinstance(fqsn, str):
+        raise ValueError(f"expect fqsn to be str but got {type(fqsn)}")
+
+    parts = fqsn.split(".")
+    if len(parts) <= 1:
+        return None
+    return parts[len(parts) - 2]
```

Comment:
Is this the same as?
`  return parts[-2]`

Reply:
Probably.